### PR TITLE
bt-642: [BUG] Fix Talent onboarding navigation on app launch

### LINKED ISSUE #642: [BUG] Fix Talent onboarding navigation on app launch
Steps to reproduce:
1. As a new user fill step 1
2. Click next
3. Kill the app
4. Relaunch the app

Actual Result:
User is again on step 1, also there is no loader or overlay view on data loading 

Expected Result:
User should appear on the last available step (step 2 for this example), 
While user data loading user sees a loader or overlay view (loader will be better)

Visual Proof (screenshots, videos, text):

https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/103518438/2065ff74-c52a-479a-8073-449cb990eadf

### LINKED ISSUE #692: [BUG] Next button doesn't redirect from step 1 to step 2
**Description:**
Next button doesn't redirect from step 1 to step 2

**Preconditions:**
The BSA talent app is open

**Environment:**
Android 13
Samsung Galaxy A52 (2340x1080)

**Steps to reproduce:**
1. Sign up as a talent
2. Fill out step 1
3. Click on the "Next" button

**Actual Result:**
Next button doesn't redirect from step 1 to step 2
Next button redirects from step 2 to step 1

**Expected Result:**
User is redirected from step 1 to step 2 when clicking on the Next button
User is redirected from step 2 to step 3 when clicking on the Next button

**Visual Proof (screenshots, videos, text):**
https://drive.google.com/file/d/1O65ZM15suLPZfWZDV32NtVhDllO06g5s/view?usp=sharing

**Severity:**
Critical (S2)

### PR DESCRIPTION
**Pull request by likeri29**

### CODE REVIEW

File: mobile/src/navigations/root/root.tsx [hunk 1]
```diff
@@ -56,8 +59,20 @@ const Root: React.FC = () => {
         if (!currentUserData) {
             return;
         }
+        const nextStep = getNextStep(completedStep);
         void dispatch(getUserDetails({ userId: currentUserData.id }));
-    }, [currentUserData, currentUserData?.id, dispatch]);
+
+        if (nextStep && !isProfileComplete) {
+            navigate(nextStep);
```
Comment (likeri29) [inline mobile/src/navigations/root/root.tsx]: Here I get an error from TS. 
`No overload matches this call.
  Argument of type '[string]' is not assignable to parameter of type 'never'.
  Overload 2 of 2, '(options: never): void', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'never'.`

Comment (katerynakostikova) [reply mobile/src/navigations/root/root.tsx]: Let's not navigate, let's set initial route in options depending on last completes step

Comment (likeri29) [reply mobile/src/navigations/root/root.tsx]: I tried to pass this nextStep to the talent onboarding, but I can't pass it without navigate. And when I use navigate I receive the same error. Maybe it'd be better if I just use type assertion `nextStep as never`?

Comment (katerynakostikova) [reply mobile/src/navigations/root/root.tsx]: You can access store in onboarding nav

**Review by katerynakostikova — APPROVED** [review]

File: mobile/src/navigations/root/root.tsx (hunk 1 above)
Comment (likeri29) [reply mobile/src/navigations/root/root.tsx]: I made it, when I log nextStep it logs it, but initialROuteName stays as it was

Comment (katerynakostikova) [reply mobile/src/navigations/root/root.tsx]: hmm then... let's add not `navigate`, but `reset` in **talent navigator**

Comment (likeri29) [reply mobile/src/navigations/root/root.tsx]: had to delete isPendingOnboardingData in the root because with it, onboarding is not working properly

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**